platform/ios/src/MGLMapView.mm

```diff
@@ -570,6 +577,13 @@ - (void)commonInit
     [self addGestureRecognizer:_rotate];
     _rotateEnabled = YES;
 
+    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+    if ([defaults objectForKey:MGLRotationThresholdWhileZooming]) {
+        _rotationThresholdWhileZooming = [[defaults objectForKey:MGLRotationThresholdWhileZooming] floatValue];
+    } else {
+        _rotationThresholdWhileZooming = 0;
+    }
```

Comment:
Doing this in `commonInit` means that changes to this key after map initialization will not take effect. There are a couple solutions that would enable it to be updated later, but I think the preferred one is also the easiest: just always query for this key in the gesture method (and don’t cache it).

[From David Smith, an Apple engineer responsible for `NSUserDefaults`](http://dscoder.com/defaults.html):

> NSUserDefaults is intended for relatively small amounts of data, queried very frequently, and modified occasionally.

Reply:
Secondarily, I’d like to see all new code related to the rotation threshold wrapped in a check — otherwise this feature flag merely disables the functionality but still goes down untested code paths for all users.

Reply:
Thank you! I think I've addressed these suggestions.